src/odemis/acq/test/acq_test.py

```diff
@@ -564,6 +565,129 @@ def on_progress_update(self, future, start, end):
         self.end = end
         self.updates += 1
 
+
+class SPARCEBICTestCase(unittest.TestCase):
+    """
+    Tests to be run with a simulated SPARC that has EBIC
+    """
+    backend_was_running = False
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            testing.start_backend(SPARC_EBIC_CONFIG)
+        except IOError as exp:
+            logging.error(str(exp))
+            raise
+
+        # Find CCD & SEM components
+        cls.microscope = model.getMicroscope()
+        cls.ebeam = model.getComponent(role="e-beam")
+        cls.sed = model.getComponent(role="se-detector")
+        cls.cld = model.getComponent(role="cl-detector")
+        cls.ebd = model.getComponent(role="ebic-detector")
+
+    def test_valid_folding(self):
+        """
+        Test folding of 3 streams that have the same roi, dwellTime and repetition values but
+        different detector names
+        """
+        # Create the streams
+        sems = stream.SEMStream("test sem", self.sed, self.sed.data, self.ebeam,
+                                emtvas={"dwellTime", "scale", "magnification", "pixelSize"})
+
+        ebics = stream.EBICSettingsStream("test ebic", self.ebd, self.ebd.data, self.ebeam)
+
+        cls = stream.CLSettingsStream("test cl", self.cld, self.cld.data, self.ebeam)
+
+        sms_ebic = stream.SEMMDStream("sem ebic", [sems, ebics])
+        sms_cl = stream.SEMMDStream("sem cl", [sems, cls])
+
+        # SEM survey settings are via the current hardware settings
+        self.ebeam.dwellTime.value = self.ebeam.dwellTime.range[0]
+
+        cls.roi.value = (0, 0, 1, 1)
+        cls.repetition.value = (1024, 1024)
+
+        ebics.roi.value = (0, 0, 1, 1)
+        ebics.repetition.value = (1024, 1024)
+
+        folds = acqmng.foldStreams([sms_ebic, sms_cl])
+        folds = tuple(folds)
+        str_cnt = len(folds)
+
+        self.assertEqual(str_cnt, 1)
```

Comment:
`str_cnt` sounds like "string" so it's a confusing name. I think you can avoid this variable totally though:
```suggestion
        self.assertEqual(str_cnt, len(folds))
```